### PR TITLE
Consolidate button

### PR DESCRIPTION
**Pull request by iantrich**
This PR updates the `service-button` element to a `button` element, with all the same functionality as the `entity-button` card. The `entity-button` card has been renamed to `button` card and just wraps the `button` element now as the setting of an `entity` is now optional.

If accepted, I plan on updating my outstanding documentation PR related to tap/hold actions: https://github.com/home-assistant/home-assistant.io/pull/7751

https://github.com/home-assistant/ui-schema/issues/211

### CODE REVIEW

File: src/common/entity/compute_state_domain.ts [hunk 1]
```diff
@@ -1,6 +1,9 @@
 import { HassEntity } from "home-assistant-js-websocket";
 import computeDomain from "./compute_domain";
 
-export default function computeStateDomain(stateObj: HassEntity) {
+export default function computeStateDomain(stateObj: HassEntity | undefined) {
+  if (!stateObj) {
```
Comment (balloob) [inline src/common/entity/compute_state_domain.ts]: We shouldn't put this check in here. It should not be called if there is no state obj.

**Comment by balloob** [comment]
> Still need to do a bit more testing

So wise.

File: src/common/entity/compute_state_name.ts [hunk 2]
```diff
@@ -5,7 +5,10 @@ type CachedDisplayEntity = HassEntity & {
   _entityDisplay?: string;
 };
 
-export default function computeStateName(stateObj: HassEntity) {
+export default function computeStateName(stateObj: HassEntity | undefined) {
+  if (!stateObj) {
```
Comment (balloob) [inline src/common/entity/compute_state_name.ts]: Same.

**Comment by balloob** [comment]
Can you include some description what this PR does?

**Comment by iantrich** [comment]
Done testing. Side note, after this gets merged, need to go and update the tap/hold gallery demos

**Pull request closed without merging**